servers/rendering/renderer_rd/storage_rd/texture_storage.cpp

```diff
@@ -1414,6 +1586,11 @@ void TextureStorage::texture_proxy_update(RID p_texture, RID p_proxy_to) {
 	}
 }
 
+// Output textures in p_textures must ALL BE THE SAME SIZE
+void TextureStorage::texture_drawable_blit_rect(const TypedArray<RID> &p_textures, const Rect2i &p_rect, RID p_material, const Color &p_modulate, const TypedArray<RID> &p_source_textures, int p_to_mipmap) {
+	// TODO: Write this code for Renderer_RD
```

Comment:
So for RD this is a little more involved than with OpenGL but the functionality already exists. If I recall correctly it's a matter of adding some extra logic when initialising the texture to create texture views for each miplayer.

Then here you can loop through the layers and use the copyeffect to scale down each layer. There might even be a function that already handles this loop for you. I'll see if I can find an example of this, pretty sure there are a few examples around the radiance map code.